app/src/sharedTest/java/org/oppia/android/app/testing/TextInputInteractionViewTestActivityTest.kt

```diff
@@ -0,0 +1,275 @@
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  @Test
+  @DisableAccessibilityChecks // Disabled, as TextInputInteractionViewTestActivity is a test file and
+  // will not be used by user
+  fun testTextInput_withNoInput_hasCorrectPendingAnswerType() {
+    val activityScenario = ActivityScenario.launch(
+      TextInputInteractionViewTestActivity::class.java
+    )
+    activityScenario.onActivity { activity ->
+      val pendingAnswer = activity.textInputViewModel.getPendingAnswer()
+      assertThat(pendingAnswer.answer).isInstanceOf(InteractionObject::class.java)
+      assertThat(pendingAnswer.answer.normalizedString).isEmpty()
+    }
+  }
+
+  @Test
+  @DisableAccessibilityChecks // Disabled, as TextInputInteractionViewTestActivity is a test file and
+  // will not be used by user
+  fun testTextInput_withChar_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(
+      TextInputInteractionViewTestActivity::class.java
+    )
+    onView(withId(R.id.test_text_input_interaction_view))
+      .perform(
+        editTextInputAction.appendText(
+          "abc"
+        )
+      )
+    activityScenario.onActivity { activity ->
+      val pendingAnswer = activity.textInputViewModel.getPendingAnswer()
+      assertThat(pendingAnswer.answer).isInstanceOf(InteractionObject::class.java)
+      assertThat(pendingAnswer.answer.objectTypeCase).isEqualTo(
+        InteractionObject.ObjectTypeCase.NORMALIZED_STRING
+      )
+      assertThat(pendingAnswer.answer.normalizedString).isEqualTo("abc")
+    }
+  }
+
+  @Test
+  @Ignore("Landscape not properly supported") // TODO(#56): Reenable once landscape is supported.
+  fun testTextInput_withChar_configChange_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(
+      TextInputInteractionViewTestActivity::class.java
+    )
+    onView(withId(R.id.test_text_input_interaction_view))
+      .perform(
+        editTextInputAction.appendText(
+          "abc"
+        )
+      )
+    activityScenario.onActivity { activity ->
+      activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+    }
+    onView(withId(R.id.test_text_input_interaction_view))
+      .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+      .check(
+        ViewAssertions
+          .matches(ViewMatchers.withText("abc"))
+      )
+  }
+
+  @Test
+  @DisableAccessibilityChecks // Disabled, as TextInputInteractionViewTestActivity is a test file and
+  // will not be used by user
+  fun testTextInput_withBlankInput_submit_emptyInputErrorIsDisplayed() {
+    ActivityScenario.launch(TextInputInteractionViewTestActivity::class.java).use {
+      scrollToSubmitButton()
+      onView(withId(R.id.submit_button)).check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
```

Comment:
Can you direct imports of `ViewAssertions` and `ViewMatchers`?

Reply:
okay

Reply:
`ViewAssertions` still not directly imported.